frontend/coprs_frontend/coprs/forms.py

```diff
@@ -1416,6 +1416,45 @@ def __init__(self, api_key=None, repo_url=None, *args, **kwargs):
             self.repo_url.data = repo_url
 
 
+class AbstractSeparatedListField(wtforms.Field):
+    """
+    For API queries (uploaded JSONs), we prefer sending data as list() over
+    comma/space separated string of string items.  In Web-UI though, we prefer
+    sending comma-separated strings (easier to specify in the available UI
+    widgets, ie text-area or normal input).
+
+    Though WTForms accepts just MultiDict for both string and list variants.
+    The string-variant is just specified as one key-value pair, while lists have
+    multiple key-value pairs with multiple different values, for example
+    [("foo", "val1"), ("foo", "val2")].
+
+    We typically want to work with string representation, because it is easier
+    to store in DB.  So the task of this custom Field class is to transform the
+    possible list representation into a XYZ-separated string.
+    """
+    def process_formdata(self, valuelist):
+        # pylint: disable=attribute-defined-outside-init
+        self.data = self.transform(valuelist)
```

Comment:
Simple as this?  @FrostyX it is maybe too simple compared to the example you provided, not sure if it covers all the cases.